exporters/otlp/otlptrace/otlptracegrpc/options.go

```diff
@@ -150,6 +150,8 @@ func WithServiceConfig(serviceConfig string) Option {
 // connection. The options here are appended to the internal grpc.DialOptions
 // used so they will take precedence over any other internal grpc.DialOptions
 // they might conflict with.
+// The [grpc.WithBlock], [grpc.WithTimeout], and [grpc.WithReturnConnectionError]
```

Comment:
Is it possible to warn when users try to use one of these?

Reply:
You mean e.g. log a warning at runtime?

Reply:
1. There it is not possible to trustworthy check if user has used any of these options.
2. [`grpc.WithTimeout`](https://pkg.go.dev/google.golang.org/grpc#WithTimeout) is already deprecated so users are quite well notified that they should not use it.
3. [`grpc.WithBlock`](https://pkg.go.dev/google.golang.org/grpc#WithTimeout) and [`grpc.WithReturnConnectionError`](https://pkg.go.dev/google.golang.org/grpc#WithReturnConnectionError) has a GoDoc comment which discourages use of this options. I created https://github.com/grpc/grpc-go/issues/7095 so that users using static analysis tools would be notified when using these options.

I think there is nothing we can do better.